python/core/personality_scoring.py

```
@@ -295,128 +380,146 @@ def score_positional(self, metrics: PersonalityMetrics) -> float:
-        streak_bonus = min(16.0, metrics.forcing_streak_max * 3.0)  # Increased
-
-        # Less penalty for errors (aggressive players take risks)
-        error_penalty = (blunder_rate * 15.0) + (mistake_rate * 10.0)  # Reduced per doc
-
-        score = base + forcing_bonus - quiet_penalty + attack_bonus + streak_bonus - error_penalty
+        total = max(1, metrics.total_moves)
+        forcing_ratio = metrics.forcing_moves / total
+        quiet_ratio = metrics.quiet_moves / total
+        forcing_accuracy = (metrics.forcing_best / metrics.forcing_moves) if metrics.forcing_moves > 0 else forcing_ratio
+        check_density = metrics.checks / total
+        capture_density = metrics.captures / total
```

Ruff (0.14.1)

Local variable `capture_density` is assigned to but never used

Remove assignment to unused variable `capture_density`

(F841)